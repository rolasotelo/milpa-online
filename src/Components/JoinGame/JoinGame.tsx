import React, { useState } from "react";
import JoinGameImage from "../../../static/join-game.png";

interface Props {
  onClickJoin: (code: string) => void;
}

const JoinGame = (props: Props) => {
  const [gameCode, setGameCode] = useState("");
  return (
    <div
      className="w-80 h-96 bg-contain bg-no-repeat"
      style={{ backgroundImage: `url(${JoinGameImage})` }}
    >
      <div className="flex flex-col">
        <div className="mx-auto">
          <button
            onClick={() => {
              props.onClickJoin(gameCode);
            }}
            className="bg-white border-4 border-mexicanPink rounded-full px-2 my-4"
          >
            JOIN
          </button>
        </div>
        <div className="mx-auto">
          <input
            className="placeholder-gray-500 placeholder-opacity-100 ..."
            placeholder="Paste game code"
            value={gameCode}
            onChange={(event) => {
              setGameCode(event.target.value);
            }}
          />
        </div>
      </div>
    </div>
  );
};

export default JoinGame;
