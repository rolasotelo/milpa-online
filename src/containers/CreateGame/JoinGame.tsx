import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { PropsWithChildren } from "../../common/interfaces";

interface Props {
  onClickJoin: (code: string) => void;
}

function JoinButton(props: { onClick: () => void; text: string }) {
  const { onClick, text } = props;
  return (
    <button
      type="button"
      className="z-10  bg-button-pink mt-4 w-52 h-24  focus:outline-none focus:bg-button-pink-pressed pl-6 pt-5 focus:pl-4 focus:pt-3 text-milpaBlue-dark"
      onClick={onClick}
      style={{
        fontFamily: "goodlife-serif, sans-serif",
        fontWeight: 400,
        fontStyle: "normal",
        fontSize: "1.75rem",
      }}
    >
      {text}
    </button>
  );
}

function Layout(props: PropsWithChildren) {
  const { children } = props;
  return (
    <div className="relative z-10 w-20.38rem tablet:w-27.5rem h-32.5rem bg-join-background ring-8 ring-inset ring-milpaBlue-dark m-5">
      <div className="absolute left-3 top-3 bg-symbol-corner-1 w-8 h-8" />
      <div className="absolute right-3 top-3 bg-symbol-corner-4 w-8 h-8" />
      <div className="flex flex-col">{children}</div>
    </div>
  );
}

function JoinGame(props: Props) {
  const { onClickJoin } = props;
  const [gameCode, setGameCode] = useState("");
  const { t } = useTranslation();
  const handleClick = React.useCallback(() => {
    onClickJoin(gameCode);
  }, [onClickJoin, gameCode]);
  const handleChange = React.useCallback((event) => {
    setGameCode(event.target.value);
  }, []);
  return (
    <Layout>
      <div className="mx-auto">
        <JoinButton onClick={handleClick} text={t("play.join.button")} />
      </div>
      <div className="mx-auto">
        <input
          className="z-40 placeholder-milpaBeige-default placeholder-opacity-100 h-input-3d w-input-3d text-center bg-pink-input bg-milpaBlue-default focus:outline-none text-milpaBeige-default"
          placeholder={t("play.join.paste")}
          value={gameCode}
          onChange={handleChange}
        />
      </div>
    </Layout>
  );
}

export default JoinGame;
