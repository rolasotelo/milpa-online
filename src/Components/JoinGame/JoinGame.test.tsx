import React from "react";
import renderer from "react-test-renderer";

import JoinGame from "./JoinGame";

describe("<JoinGame />", () => {
  const defaultProps = {
    onClickJoin: (code: string) => {},
    text: "JOIN",
  };
  const wrapper = renderer.create(<JoinGame {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
