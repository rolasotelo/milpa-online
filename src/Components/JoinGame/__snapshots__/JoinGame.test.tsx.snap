// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JoinGame /> render 1`] = `
<div
  className="w-20.38rem md:w-27.5rem h-32.5rem bg-join-background ring-8 ring-inset ring-black m-5"
>
  <div
    className="flex flex-col"
  >
    <div
      className="mx-auto"
    >
      <button
        className="bg-button-pink w-52 h-24 mt-4 focus:outline-none focus:bg-button-pink-pressed pl-3 pb-2 focus:pl-2 focus:pb-1"
        onClick={[Function]}
        style={
          Object {
            "fontFamily": "goodlife-sans-condensed, sans-serif",
            "fontSize": "2rem",
            "fontStyle": "normal",
            "fontWeight": 400,
          }
        }
      >
        JOIN
      </button>
    </div>
    <div
      className="mx-auto"
    >
      <input
        className="placeholder-gray-500 placeholder-opacity-100 ..."
        onChange={[Function]}
        placeholder="Paste game code"
        value=""
      />
    </div>
  </div>
</div>
`;
